### PR TITLE
Build and release

### PR DESCRIPTION
**Pull request by featheredtoast**
Github action on version tags like v1.2.3 and upload releases.

All assets will then get compiled and uploaded to the release.

Latest releases will be available at github.com/discourse/launcher/releases/latest

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -37,3 +39,50 @@ jobs:
       - name: Run tests
         working-directory: ./v2
         run: go test ./...
+
+  create_release:
```
Comment (tgxworld) [inline .github/workflows/build.yml:43]:
```suggestion
  create_github_release:
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -37,3 +39,50 @@ jobs:
...
+    runs-on: ubuntu-latest
+    needs: [lint, test]
+    steps:
+      - uses: actions/checkout@v4
+      - name: build and release
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh release create ${{ github.ref_name }}
+  build_release:
```
Comment (tgxworld) [inline .github/workflows/build.yml:56]:
```suggestion
  build_and_upload_to_github_release:
```

Abit verbose but I find this much clearer.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -37,3 +39,50 @@ jobs:
...
+    if: startsWith(github.ref, 'refs/tags/v')
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    needs: [lint, test, create_release]
+    strategy:
+      matrix:
+        goos: [linux, darwin]
+        goarch: ["386", amd64, arm64]
```
Comment (tgxworld) [inline .github/workflows/build.yml:65]: I'm not familiar with the `386` ARCH but do we need it?

**Comment by tgxworld** [comment]
@featheredtoast I think we should also create a `Makefile` that encapsulates the commands for pushing a new tag to github.

File: .github/workflows/build.yml (hunk 3 above)
Comment (featheredtoast) [reply .github/workflows/build.yml:65]: it's the 86 part of x86/64 -- IE 32 bit systems. Still usable until 32 bit integers roll over in 2038.

**Comment by tgxworld** [comment]
Thank you @featheredtoast 👍

**Pull request merged**